python/remotivelabs-broker/remotivelabs/broker/sync/client.py

```diff
@@ -0,0 +1,261 @@
+from ..generated.sync import network_api_pb2_grpc
+from ..generated.sync import system_api_pb2_grpc
+from ..generated.sync import traffic_api_pb2_grpc
+
+
+class SignalWrapper:
+    """
+    Wrapper around protobuf generated class to make it a bit simpler to use
+    to make us learn how we want the next version of the API to look like.
+
+    Use the signal.is_{type}() functions to validate type before you get value.
+    Use signal.value() to get the actual value without any validation.
+    Use signal.get_raw() to get the raw bytes
+    Use signal.{type}_value() to get a validated typed value or error if something is wrong
+    """
+
+    def __init__(self, signal: network_api.Signal):
+        self.signal: network_api.Signal = signal
+
+    def __str__(self):
+        return self.to_json()
+
+    def to_json(self) -> str:
+        return json.dumps(self.as_dict())
+
+    def is_integer(self) -> bool:
+        return self.signal.HasField("integer")
+
+    def is_double(self) -> bool:
+        return self.signal.HasField("double")
+
+    def is_arbitration(self) -> bool:
+        return self.signal.HasField("arbitration")
+
+    def is_raw(self) -> bool:
+        return self.signal.raw != b""
+
+    def get_raw(self) -> Union[bytes, None]:
+        if self.is_raw():
+            return self.signal.raw
+        else:
+            return None
+
+    def __get_value(self) -> Union[str, int, float, bool, None]:
+        if self.signal.raw != b"":
+            return "0x" + binascii.hexlify(self.signal.raw).decode("ascii")
+        elif self.signal.HasField("integer"):
+            return self.signal.integer
+        elif self.signal.HasField("double"):
+            return self.signal.double
+        elif self.signal.HasField("arbitration"):
+            return self.signal.arbitration
+        else:
+            return None
+
+    def timestamp_us(self):
+        return self.signal.timestamp
+
+    def name(self):
+        return self.signal.id.name
+
+    def namespace(self):
+        return self.signal.id.namespace.name
+
+    def value(self):
+        return self.__get_value()
+
+    def float_value(self):
+        v = self.__get_value()
+        if isinstance(v, float):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type float but got {type(v)}")
+
+    def int_value(self):
+        v = self.__get_value()
+        if isinstance(v, int):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type int but got {type(v)}")
+
+    def bool_value(self):
+        v = self.__get_value()
+        if isinstance(v, bool):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type bool but got {type(v)}")
+
+    def bytes_value(self):
+        v = self.__get_value()
+        if isinstance(v, bytes):
+            return v
+        else:
+            raise BrokerException(f"{v} was not expected type bytes but got {type(v)}")
```

Comment:
Duplicadet logic. I suggest out shared logic into other function.

Reply:
@carlmartus How do I accomplish that i python? I tried but could not figure out how to send the type to a separate function

Reply:
Fixed